skbio/tree/_tree.py

```diff
@@ -617,6 +617,77 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of nodes in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(len(path))
+        3
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(len(path_2))
+        5
+
+        """
+        # check to see if nodes are on same tree
+        root1, root2 = self.root(), other.root()
```

Comment:
Both `root` and `ancestor` involves traversing from the current node to the root. If a tree has n taxa, the time complexity of traversing to the root would be $O(log_2 n)$, which is expensive. Therefore, rather than calling `root` and `ancestor` separately, you may only call `ancestor`, the last node of which must be the root.

Reply:
Additionally, you don't even need to explicitly compare `anc1[-1] is anc2[-1]`. The `for` loop below already covers this. If it doesn't produce an `lca`, then the two nodes are not connected.